helper/src/commonMain/kotlin/com/algolia/instantsearch/helper/hierarchical/HierarchicalConnectionFilterState.kt

```diff
@@ -12,30 +10,30 @@ import com.algolia.search.model.filter.Filter
 internal data class HierarchicalConnectionFilterState(
     private val viewModel: HierarchicalViewModel,
     private val filterState: FilterState
-): ConnectionImpl() {
-
-    private val filterGroupID = FilterGroupID(hierarchicalGroupName, FilterOperator.And)
+) : ConnectionImpl() {
 
     private val updateSelections: Callback<Filters> = { filters ->
-        viewModel.selections.value = (filters.getFacetFilters(filterGroupID).first().value as Filter.Facet.Value.String).raw.split(viewModel.separator)
+        val hierarchicalFilter = filters.getHierarchicalFilters(viewModel.attribute)
+        val hierarchicalValue = hierarchicalFilter?.filter?.value as Filter.Facet.Value.String
+
+        viewModel.selections.value = hierarchicalValue.raw.split(viewModel.separator)
```

Comment:
Isn't `hierarchicalValue` nullable here, as it's equal to `hierarchicalFilter?.foo?.bar`?

```suggestion
        viewModel.selections.value = hierarchicalValue?.raw?.split(viewModel.separator)
```

Reply:
Fixed.